File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        
        Dictionary<int, int> dic = new Dictionary<int, int>();
        
        for(int i=0; i<nums.Length; i++)
        {
            if(dic.TryGetValue(target - nums[i], out int r)){
                return new int[2]{i, r};
            }
            //dic.Add(nums[i], i);
            dic[nums[i]] = i;     
        }
        
        return new int[2];
    }
}

File: 0001-two-sum/0001-two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    //if(number.length < 2 || number = undefined) return new Array[]: number;
    const newMap = new Map();
    
    for(let i=0; i < nums.length; i++){
        const remaining = target - nums[i];
        if(newMap.has(target-nums[i])){
            return [newMap.get(remaining), i];
        }
        newMap.set(nums[i], i);
    }
    
};